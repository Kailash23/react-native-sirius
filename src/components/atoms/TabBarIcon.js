import React from 'react';
import Ionicons from 'react-native-vector-icons/Ionicons';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import { withTheme } from 'react-native-paper';
import { layout } from '@utils/Layout';

function TabBarIcon(props) {
  const { colors } = props.theme;
  if (props.homeIcon) {
    return (
      <FontAwesome5
        name={'newspaper'}
        solid
        size={layout.isMobile ? 26 : 22}
        style={{ marginBottom: -3 }}
        color={props.focused ? colors.primary : colors.tabIcon}
      />
    );
  } else {
    return (
      <Ionicons
        name={props.name}
        size={26}
        style={{ marginBottom: -3 }}
        color={props.focused ? colors.primary : colors.tabIcon}
      />
    );
  }
}

export default withTheme(TabBarIcon);
