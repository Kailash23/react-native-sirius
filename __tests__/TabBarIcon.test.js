import TabBarIcon from '@atoms/TabBarIcon';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
const moment = require('moment');

configure({ adapter: new Adapter() });

describe('Testing <TabBarIcon/>', () => {
  test('TabBarIcon should return settings Ionicons ', () => {
    const wrapper = shallow(<TabBarIcon name={'md-settings'} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('TabBarIcon should return home FontAwesome icon ', () => {
    const wrapper = shallow(<TabBarIcon homeIcon={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
