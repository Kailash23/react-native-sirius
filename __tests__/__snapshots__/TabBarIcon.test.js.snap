// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TabBarIcon/> TabBarIcon should return home FontAwesome icon  1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  homeIcon={true}
/>
`;

exports[`Testing <TabBarIcon/> TabBarIcon should return settings Ionicons  1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  name="md-settings"
/>
`;
